docs: MkDocs index and nav redesign

cwhite911 opened the pull request:
- Completely redesign index page to be Getting started page with overview and just couple links elsewhere.
- Add intro pages for interfaces (for now, getting started with interface and overview of interface are the same pages).
- Change navigation to highlight interfaces, concepts, and tools.

This is a first draft, and several issues should be addressed later. Besides the content, links to addons and libpython need a revision.

Related to issue #5162. (Originally raised as index page redesign, but index page does not have to make up for a menu anymore because we do have a menu with MkDocs.)

![image](https://github.com/user-attachments/assets/6d790c85-ac15-44dc-aa09-7505345b0a0d)

wenzeslaus commented on man/mkdocs/index.md:
```diff
@@ -0,0 +1,55 @@
+---
+title: GRASS GIS 8.5.0dev Reference Manual
```
Limit hardcoded versions as much as possible.

wenzeslaus commented on man/mkdocs/index.md:
```diff
@@ -0,0 +1,55 @@
+---
+
+
+## **What is GRASS?**
```
I'm surprised this passes linting, but anyway, is `**` needed?

wenzeslaus commented on man/mkdocs/mkdocs.yml:
```diff
@@ -26,6 +26,8 @@ theme:
     - search.suggest
     - search.highlight
     - content.code.copy
+    - content.tabs.link
```
Can you please separate the tabs into a new PR to get it merged quickly, including the linked tabs option here? I was able to enable the tabs (locally), but the doc (mis)lead me to include some custom JS to keep the tabs in sync and that was not working for me.

cwhite911 replied:
https://github.com/OSGeo/grass/pull/5277

wenzeslaus commented on doc/development/Makefile:
```diff
@@ -0,0 +1,9 @@
+
+include $(MODULE_TOPDIR)/include/Make/Other.make
+
+# So far, we disntinguished user and contributor documentation here by
+# extension. This is no longer possible with Markdown.
```
This is really relevant only for the parent dir, not here.

echoix replied:
```suggestion
# So far, we distinguished user and contributor documentation here by
# extension. This is no longer possible with Markdown.
```

wenzeslaus commented on doc/development/Makefile:
```diff
@@ -0,0 +1,9 @@
+
+# So far, we disntinguished user and contributor documentation here by
+# extension. This is no longer possible with Markdown.
+MDFILES := github_guide.md style_guide.md
```
Let's move this whole Makefile and the task of publishing development-related Markdown files to another PR or issue.

wenzeslaus commented on doc/quickstart.md:
Needs to be removed more. (?)

wenzeslaus commented on man/mkdocs/index.md:
````diff
@@ -0,0 +1,121 @@
+```bash
+
+grass -c EPSG:3385 {project directory} --gtext
+
+Starting GRASS GIS...
+
+          __________  ___   __________    _______________
+         / ____/ __ \/   | / ___/ ___/   / ____/  _/ ___/
+        / / __/ /_/ / /| | \__ \\_  \   / / __ / / \__ \
+       / /_/ / _, _/ ___ |___/ /__/ /  / /_/ // / ___/ /
+       \____/_/ |_/_/  |_/____/____/   \____/___//____/
+
+Welcome to GRASS GIS 8.5.0dev (5ffc9ea669)
+GRASS GIS homepage:                      https://grass.osgeo.org
+This version running through:            Bash Shell (/bin/bash)
+Help is available with the command:      g.manual -i
+See the licence terms with:              g.version -c
+See citation options with:               g.version -x
+If required, restart the GUI with:       g.gui wxpython
+When ready to quit enter:                exit
````
Maybe little too many lines? (That output should be shorter, but even a shorter one would still be little too long for here.)

cwhite911 replied:
I've change the section.

wenzeslaus commented on man/mkdocs/index.md:
````diff
@@ -0,0 +1,121 @@
+
+```bash
+
+grass -c EPSG:3385 {project directory} --gtext
````
```suggestion
grass -c EPSG:3385 {project directory} --text
```

For true no-GUI experience.

wenzeslaus commented on man/mkdocs/index.md:
````diff
@@ -0,0 +1,121 @@
+r.slope.aspect elevation=elevation slope=slope aspect=aspect
+```
+
+### Python Scripts
+
+The `grass.script` module provides a Python interface to GRASS. This allows
+users to write Python scripts to interact with GRASS. The `grass.script` module
````
We need to start describing the in-session version grass.script.setup.init.

wenzeslaus commented on man/mkdocs/index.md:
````diff
@@ -0,0 +1,121 @@
+
+```
+
+Now you can run GRASS commands in the terminal.
+
+```sh
+g.region raster=elevation
+r.slope.aspect elevation=elevation slope=slope aspect=aspect
+```
````
There is an alternative to this and that's `--exec` which I would say is popular for certain use cases.

wenzeslaus commented on man/mkdocs/index.md:
```diff
@@ -0,0 +1,121 @@
+
+Add Content here.
+
+## Processing Tools
+
+### GRASS Tool Prefixes
+
+| Prefix | Category                         | Description                        | Link                                      |
+|--------|----------------------------------|------------------------------------|-------------------------------------------|
+| `g.`   | General                          | General GIS management tools       | [General Tools](general.md)               |
```
Just make this presentation of tool, rather than prefixes.

wenzeslaus commented on man/mkdocs/index.md:
```diff
@@ -0,0 +1,121 @@
+
+Add Content here.
+
+## Processing Tools
```
Add link to addons to the end of the section.

wenzeslaus commented on man/mkdocs/index.md:
````diff
@@ -0,0 +1,121 @@
+r.slope.aspect elevation=elevation slope=slope aspect=aspect
+```
+
+### Python Scripts
````
C doc and possibly general tool writing stuff (standard parser options): I wonder if to put it here as sort of next steps ("is this not enough?") or at the end to a separate section. This would be the usage as a modeling platform.

wenzeslaus reviewed (commented):
I like the approach!

cwhite911 commented:
Everything here is still a WIP, but here is a snapshot of the current status.

## Getting Started
![image](https://github.com/user-attachments/assets/b1b6bade-e63a-424a-a390-ace5537d72dd)

## Terminal Overview
![image](https://github.com/user-attachments/assets/f60c6cf7-19f6-4677-820b-1eb7ca9bbcc5)

## Python Overview
![image](https://github.com/user-attachments/assets/6aec3d52-40fc-45bc-b5e1-a0ab51ac55ad)

## Jupyter Overview
![image](https://github.com/user-attachments/assets/b040d7fe-a409-4f7b-9e99-6757511c4462)

## Development
![image](https://github.com/user-attachments/assets/6db8e04c-1f7a-48b7-92ff-f389f95ad3b5)

petrasovaa commented on man/mkdocs/index.md:
```diff
@@ -0,0 +1,155 @@
+
+## What is GRASS?
+
+[GRASS](https://grass.osgeo.org/) is a geosptial processing engine for
+advance analysis and visualization of geospatial data. It is a powerful tool for
+processing and analyzing geospatial data sets. GRASS is a free and open source
+software, released under an open source [GNU GPLed](https://www.gnu.org/licenses/gpl.html).
```
```suggestion
[GRASS](https://grass.osgeo.org/) is a geospatial processing engine for
advance analysis and visualization of geospatial data. GRASS is a free and open source
software, released under the [GNU GPL](https://www.gnu.org/licenses/gpl.html).
```

petrasovaa commented on man/mkdocs/index.md:
```diff
@@ -0,0 +1,155 @@
+The terminal interface allows you to start a GRASS session to run GRASS
+commands, execute scripts, or open the GUI.
+
+Here we creating a new project for the NAD83(HARN)/North Carolina coordinate
```
```suggestion
Here we create a new project for the NAD83(HARN)/North Carolina coordinate
```

petrasovaa commented on man/mkdocs/index.md:
```diff
@@ -0,0 +1,155 @@
+processing and analyzing geospatial data sets. GRASS is a free and open source
+software, released under an open source [GNU GPLed](https://www.gnu.org/licenses/gpl.html).
+
+Downloaded and installed GRASS here.
+
+<!-- markdownlint-disable-next-line MD013 -->
+[:material-download: Download and Install](https://grass.osgeo.org/download/){ .md-button }
```
```suggestion
<!-- markdownlint-disable-next-line MD013 -->
[:material-download: Download GRASS](https://grass.osgeo.org/download/){ .md-button }
```

petrasovaa commented on man/mkdocs/index.md:
```diff
@@ -0,0 +1,155 @@
+
+### Terminal
+
+The terminal interface allows you to start a GRASS session to run GRASS
```
```suggestion
The command line interface allows you to start a GRASS session to run GRASS
```

cwhite911 commented:
Some more updated screenshots.

## Getting Started

- New Support GRASS button

![image](https://github.com/user-attachments/assets/58d14e94-87a2-4bc3-bc19-c1345c35736d)

## Python Overview

- Added grass.scripts section

![image](https://github.com/user-attachments/assets/32efeeb6-7461-4069-a634-c679e0047a5c)


## Jupyter Notebooks

- Added more text and images

![image](https://github.com/user-attachments/assets/8c2a2c9e-0c77-4de7-b2c7-5558caf59c0e)

wenzeslaus commented:
I have nav updates I want to discuss and push to the branch, let's meet.

wenzeslaus commented:
![image](https://github.com/user-attachments/assets/2aab22e8-af6d-4e48-b0e6-3df8fc86945f)
![image](https://github.com/user-attachments/assets/b22bb425-f9f1-4bd4-bedb-d33e86bd164e)
![image](https://github.com/user-attachments/assets/0a1abeca-5d31-411d-93ef-83c2d4c2c5ef)
![image](https://github.com/user-attachments/assets/46c1e0cf-ae25-48ed-9143-45c0b0cb841d)

cwhite911 commented:
Here is a snapshot from the Python section. 

TODOs
- grass.scripts Raster3d, Database
- grass.pygrass: vector and modules are still in progress.
- General clean up: typos, grammer, links
- Design: Finalize layout
- Optional: Improve or remove flow chart at the begin

## Python

![image](https://github.com/user-attachments/assets/ad874611-4134-40b1-bb47-cd9fabdb876f)

wenzeslaus reviewed (commented):
I improved things like command line and the main intro, but only fixed issues in the large Python documents. Given this is a first draft, I removed the images (2MB after further compression) to save repo space because the images may or may not be used depending on the rewrite. The Python document is little too long (800+ lines of Markdown), so that's one of the things we may want to address as a next step. The Mermaid blocks work on GitHub as well, so the experience is smooth and it's not a big commitment to specific dependency.
